### PR TITLE
feat(campaign): initialize `CampaignChains` structure

### PR DESCRIPTION
**Pull request by lubtd**
`CampaignChains` allows to retrieve all the chains associated to a campaign

### CODE REVIEW

File: x/campaign/types/key_campaign_chains.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const (
+	// CampaignChainsKeyPrefix is the prefix to retrieve all CampaignChains
+	CampaignChainsKeyPrefix = "CampaignChains/value/"
+)
+
+// CampaignChainsKey returns the store key to retrieve a CampaignChains from the index fields
+func CampaignChainsKey(
+	campaignID uint64,
+) []byte {
```
Comment (Pantani) [inline x/campaign/types/key_campaign_chains.go:15]:
```suggestion
func CampaignChainsKey(campaignID uint64) []byte {
```

**Review by Pantani — DISMISSED** [review]

File: proto/campaign/genesis.proto [hunk 2]
```diff
@@ -2,10 +2,13 @@ syntax = "proto3";
...
 // this line is used by starport scaffolding # genesis/proto/import
+import "gogoproto/gogo.proto";
+import "campaign/campaign_chains.proto";
 
 option go_package = "github.com/tendermint/spn/x/campaign/types";
 
 // GenesisState defines the campaign module's genesis state.
 message GenesisState {
     // this line is used by starport scaffolding # genesis/proto/state
+		repeated CampaignChains campaignChainsList = 1 [(gogoproto.nullable) = false]; // this line is used by starport scaffolding # genesis/proto/stateField
```
Comment (Pantani) [inline proto/campaign/genesis.proto:13]:
```suggestion
    repeated CampaignChains campaignChainsList = 1 [(gogoproto.nullable) = false]; // this line is used by starport scaffolding # genesis/proto/stateField
```

File: proto/campaign/query.proto [hunk 3]
```diff
@@ -4,12 +4,27 @@ package tendermint.spn.campaign;
...
 option go_package = "github.com/tendermint/spn/x/campaign/types";
 
 // Query defines the gRPC querier service.
 service Query {
     // this line is used by starport scaffolding # 2
+
+	// Queries a campaignChains by index.
+	rpc CampaignChains(QueryGetCampaignChainsRequest) returns (QueryGetCampaignChainsResponse) {
+		option (google.api.http).get = "/tendermint/spn/campaign/campaignChains/{campaignID}";
+	}
```
Comment (Pantani) [inline proto/campaign/query.proto:19]:
```suggestion
    // Queries a campaignChains by index.
    rpc CampaignChains(QueryGetCampaignChainsRequest) returns (QueryGetCampaignChainsResponse) {
        option (google.api.http).get = "/tendermint/spn/campaign/campaignChains/{campaignID}";
    }
```

File: proto/campaign/query.proto [hunk 4]
```diff
@@ -4,12 +4,27 @@ package tendermint.spn.campaign;
...
 }
 
 // this line is used by starport scaffolding # 3
+message QueryGetCampaignChainsRequest {
+	  uint64 campaignID = 1;
```
Comment (Pantani) [inline proto/campaign/query.proto:24]:
```suggestion
    uint64 campaignID = 1;
```

File: proto/campaign/query.proto [hunk 5]
```diff
@@ -4,12 +4,27 @@ package tendermint.spn.campaign;
...
+
+}
+
+message QueryGetCampaignChainsResponse {
+	CampaignChains campaignChains = 1 [(gogoproto.nullable) = false];
```
Comment (Pantani) [inline proto/campaign/query.proto:29]:
```suggestion
    CampaignChains campaignChains = 1 [(gogoproto.nullable) = false];
```

File: x/campaign/keeper/campaign_chains.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+	store.Set(types.CampaignChainsKey(
+		campaignChains.CampaignID,
+	), b)
+}
+
+// GetCampaignChains returns a campaignChains from its index
+func (k Keeper) GetCampaignChains(
+	ctx sdk.Context,
+	campaignID uint64,
+
```
Comment (Pantani) [inline x/campaign/keeper/campaign_chains.go:22]:
```suggestion
```

File: x/campaign/keeper/campaign_chains_test.go [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+// Prevent strconv unused error
+var _ = strconv.IntSize
+
+func createNCampaignChains(keeper *campaignkeeper.Keeper, ctx sdk.Context, n int) []types.CampaignChains {
+	items := make([]types.CampaignChains, n)
+	for i := range items {
+		items[i].CampaignID = uint64(i)
+
```
Comment (Pantani) [inline x/campaign/keeper/campaign_chains_test.go:21]:
```suggestion
```

File: proto/campaign/genesis.proto (hunk 2 above)
Comment (lubtd) [reply proto/campaign/genesis.proto:13]: Seems like we have some inconsistencies in generated proto files: https://github.com/tendermint/starport/issues/1561

**Review by Pantani — DISMISSED** [review]

**Review by ilgooz — DISMISSED** [review]

**Pull request merged**